go/fn/object_test.go

```diff
@@ -237,6 +238,38 @@ func TestSetNestedFields(t *testing.T) {
 	}
 }
 
+func TestInternalAnnotationsUntouchable(t *testing.T) {
+	o := NewEmptyKubeObject()
+	// Verify the "upstream-identifier" annotation cannot be changed via SetNestedStringMap
+	o.SetNestedStringMap(map[string]string{"owner": "kpt"}, "metadata", "annotations")
+	if stringMapVal := o.NestedStringMapOrDie("metadata", "annotations"); !reflect.DeepEqual(stringMapVal, map[string]string{"owner": "kpt"}) {
+		t.Errorf("annotations cannot be set via SetNestedStringMap, got %v", stringMapVal)
+	}
+	err := o.SetNestedStringMap(map[string]string{UpstreamIdentifier: "|Deployment|default|dp"}, "metadata", "annotations")
```

Comment:
Group is empty. Should it be `apps`?

Reply:
haha, sure. Thanks! FYI, we do accept empty group, for core v1 resources.